src/java/org/apache/cassandra/utils/ImmutableUniqueList.java

```diff
@@ -19,24 +19,39 @@
 package org.apache.cassandra.utils;
 
 import java.util.AbstractList;
-import java.util.AbstractSet;
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Iterator;
 import java.util.List;
 import java.util.RandomAccess;
 
 import org.agrona.collections.Object2IntHashMap;
 
-public class ImmutableUniqueList<T> extends AbstractList<T> implements RandomAccess
```

Comment:
this was in `test/unit` as i used this for tests, but moved it over to `src` to leverage.  I also took the opportunity to cleanup a few things